app/actions/local/channel.test.ts

```diff
@@ -0,0 +1,192 @@
+
+        const {models, error} = await switchToChannel(serverUrl, channelId, teamId);
+
+        const systemValues = await queryCommonSystemValues(operator.database);
+        const teamHistory = await queryTeamHistory(operator.database);
+        const channelHistory = await queryChannelHistory(operator.database, teamId);
+        const member = await queryMyChannel(operator.database, channelId);
+
+        expect(error).toBe(undefined);
+        expect(models?.length).toBe(5); // Viewed at, channel history, team history, currentTeamId and currentUserId
+        expect(systemValues.currentTeamId).toBe(teamId);
+        expect(systemValues.currentChannelId).toBe(channelId);
+        expect(teamHistory.length).toBe(1);
+        expect(teamHistory[0]).toBe(teamId);
+        expect(channelHistory.length).toBe(1);
+        expect(channelHistory[0]).toBe(channelId);
+        expect(member?.lastViewedAt).toBe(now);
+        expect(dismissAllModalsAndPopToScreen).toHaveBeenCalledTimes(1);
+    });
+
+    it('switch to dm channel in different team', async () => {
+        const tChannel = {...channel, team_id: ''};
+        await operator.handleChannel({channels: [tChannel], prepareRecordsOnly: false});
+        await operator.handleMyChannel({channels: [tChannel], myChannels: [channelMember], prepareRecordsOnly: false});
+        await operator.handleSystem({systems: [{id: SYSTEM_IDENTIFIERS.CURRENT_TEAM_ID, value: 'currentTeamId'}, {id: SYSTEM_IDENTIFIERS.CURRENT_CHANNEL_ID, value: 'currentChannelId'}], prepareRecordsOnly: false});
+
+        const {models, error} = await switchToChannel(serverUrl, channelId, teamId);
+
+        const systemValues = await queryCommonSystemValues(operator.database);
+        const teamHistory = await queryTeamHistory(operator.database);
+        const channelHistory = await queryChannelHistory(operator.database, teamId);
+        const member = await queryMyChannel(operator.database, channelId);
+
+        expect(error).toBe(undefined);
+        expect(models?.length).toBe(5); // Viewed at, channel history, team history, currentTeamId and currentUserId
+        expect(systemValues.currentTeamId).toBe(teamId);
+        expect(systemValues.currentChannelId).toBe(channelId);
+        expect(teamHistory.length).toBe(1);
+        expect(teamHistory[0]).toBe(teamId);
+        expect(channelHistory.length).toBe(1); // Currently errors since it is being stored in the current team history and not the destination history
+        expect(channelHistory[0]).toBe(channelId);
+        expect(member?.lastViewedAt).toBe(now);
+        expect(dismissAllModalsAndPopToScreen).toHaveBeenCalledTimes(1);
+    });
+
+    it('switch to a channel from different team without passing the teamId', async () => {
+        // TODO: Why do we need to pass the team id?
```

Comment:
is really because of this https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/actions/remote/entry/notification.ts#L112 the push notification entry point

Reply:
But we don't need it, right? If the channel is a DM/GM, we are taking the current team id either way, and if the channel is a normal channel, we take the team id from the notification, that should be the same team id as the one stored in the channel, right? So we could simplify the logic in this function (and the tests) by removing the teamId completely.

Reply:
probably.. I think it should be fine, but I would love to look again into the notification entry and identify the use case where the teamId is required

Reply:
I found a reason to keep it:

When we change teams, we want to switch the team and go to the last channel visited in that team. That last channel may be a DM. If we don't allow to change both channel and team, we end up in weird behaviors.

So... the question is... if the channel belongs to a team, what is the expected behaviour? Ignore the teamId passed?